nav2_costmap_2d/src/footprint_subscriber.cpp

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2019 Intel Corporation
```

Comment:
I see no issue with this, but I would believe that having this read the configuration file and it being a ROS agnostic helper to read from configuration file and getting as necessary may be more clear since somewhere else inside of costmap_2d we read the footprint and then publish it. Seems odd to have things running in the same process in the same node talking to each other over ROS to share data

Reply:
I believe in `Costmap2DROS`, the footprint subscription is listening on a Polygon topic in order to update the footprint across all layers via `setUnpaddedFootprint` in its callback. Since in this class we're listening on the published PolygonStamped topic and have a default callback to convert and store the robot oriented footprint, I see them as separate use cases.